ceph_salt/core.py

```diff
@@ -44,18 +45,21 @@ def ipsv6(self):
 
     @property
     def public_ip(self):
+        def _is_loopback(addr):
+            return ipaddress.ip_address(addr).is_loopback
+
         if self._public_ip is None:
             result = GrainsManager.get_grain(self.minion_id, 'fqdn_ip4')
             _public_ip = result[self.minion_id][0]
-            if _public_ip == '127.0.0.1':
-                logger.debug('fqdn_ipv4 grain is 127.0.0.1, falling back to ipv4 grain')
+            if _is_loopback(_public_ip):
+                logger.debug('fqdn_ipv4 grain is a loopback address, falling back to ipv4 grain')
                 result = GrainsManager.get_grain(self.minion_id, 'ipv4')
                 for addr in result[self.minion_id]:
-                    if addr != '127.0.0.1':
+                    if not _is_loopback(addr):
                         _public_ip = addr
                         break
-                if _public_ip == '127.0.0.1':
-                    logger.warning("'%s' public IP is the loopback interface IP ('127.0.0.1')",
```

Comment:
Would be great if you add _public_ip value to the log instead of 127.0.0.1

Reply:
yeah good point, I've re-added the _public_ip to the log message